phangsPipeline/utilsImages.py

```diff
@@ -0,0 +1,160 @@
+# Routines relates to python (not CASA) image handling.
+
+import numpy as np
+import os
+import re
+import warnings
+import math
+
+import astropy.units as u
+from astropy.io import fits
+from astropy.wcs import WCS
+from astropy.coordinates import SkyCoord
+
+# ------------------------------------------------------------------------
+# Make new headers
+# ------------------------------------------------------------------------
+
+def make_simple_header_from_box(
+        tlc_coord = None, brc_coord = None,
+        pix_scale_deg = None, pad_arcsec = 0.0):
+    """Wraps make_simple_header.
+
+    tlc_coord and brc_coord : tuples of floats or SkyCoords giving
+    corners of the image to be made.
+    
+    """
+
+    # If not SkyCoords convert
+    if isinstance(tlc_coord, SkyCoord) == False:
+        tlc_coord = SkyCoord(
+            tlc_coord[0]*u.deg, tlc_coord[1]*u.deg, frame='icrs')
+
+    if isinstance(brc_coord, SkyCoord) == False:
+        brc_coord = SkyCoord(
+            brc_coord[0]*u.deg, brc_coord[1]*u.deg, frame='icrs')
+
+    # Find the midpoint
+    pa = tlc_coord.position_angle(brc_coord)
+    sep = tlc_coord.separation(brc_coord)
+    midpoint = tlc_coord.directional_offset_by(pa, sep/2.0)
+    
+    # Find the edge
+    extent_deg = sep.data/np.sqrt(2)
+    extent_deg += 2.0*pad_arcsec/3600.
+
+    # Call the other routine
+    hdr = make_simple_header(
+        ra_ctr = midpoint.ra.deg, dec_ctr = midpoint.dec.deg,
+        extent_x = extent_deg, extent_y = extent_deg,
+        pix_scale = pix_scale_deg)
+
+    return(hdr)
+    
+def make_simple_header(
+        ra_ctr = None, dec_ctr = None, pix_scale = None,
+        extent_x = None, extent_y = None,
+        nx = None, ny = None):
+    """
+    Make a simple FITS header centered on some locations.
+
+    
+    Parameters
+    ----------
+    pix_scale : required. Size in decimal degrees of a pixel.
+
+    extent_x : the angular extent of the image along the x coordinate
+    extent_y : the angular extent of the image along the y coordinate
+
+    nx : the number of x pixels (not needed with extent_x and pix_scale)
+    ny : the number of y pixels (not needed with extent_y and pix_scale)
+    
+    """
+    
+    # Deal with center, skycoords, units, etc. better
+    
+    if nx is not None and ny is not None:
+        extent_x = pix_scale * nx
+        extent_y = pix_scale * ny
+    elif extent_x is not None and extent_y is not None:
+        nx = int(np.ceil(extent_x*0.5 / pix_scale) * 2 + 1)
+        ny = int(np.ceil(extent_y*0.5 / pix_scale) * 2 + 1)
+
+    new_hdr = fits.Header()
```

Comment:
No issues but reading into the broad public record that having pre-defined "typical" headers or WCS objects be defined would be useful!